lib/Runtime/Library/BoundFunction.cpp

```diff
@@ -136,16 +136,28 @@ namespace Js
         Var newVarInstance = nullptr;
         if (callInfo.Flags & CallFlags_New)
         {
-          if (JavascriptProxy::Is(targetFunction))
-          {
-            JavascriptProxy* proxy = JavascriptProxy::FromVar(targetFunction);
-            Arguments proxyArgs(CallInfo(CallFlags_New, 1), &targetFunction);
-            args.Values[0] = newVarInstance = proxy->ConstructorTrap(proxyArgs, scriptContext, 0);
-          }
-          else
-          {
-            args.Values[0] = newVarInstance = JavascriptOperators::NewScObjectNoCtor(targetFunction, scriptContext);
-          }
+            if (args.HasNewTarget())
+            {
+                // target has an overriden new target make a new object from the newTarget
+                Var newTargetVar = args.GetNewTarget();
+                if (!JavascriptOperators::IsConstructor(newTargetVar))
+                {
+                    JavascriptError::ThrowTypeError(scriptContext, JSERR_NeedConstructor, _u("newTarget"));
+                }
+                RecyclableObject* newTarget = RecyclableObject::UnsafeFromVar(args.GetNewTarget());
```

Comment:
nit: can reuse `newTargetVar` here

Reply:
OK - will also revise to do AssertOrFailFast(IsConstructor) instead of the runtime error per @MSLaguana's comment above.